pandas/core/dtypes/dtypes.py

```diff
@@ -334,15 +340,16 @@ def _finalize(self,
                   fastpath: bool = False,
                   ) -> None:
 
-        if ordered is not None:
+        if ordered is not None and ordered is not sentinel:
             self.validate_ordered(ordered)
 
         if categories is not None:
             categories = self.validate_categories(categories,
                                                   fastpath=fastpath)
 
         self._categories = categories
-        self._ordered = ordered
+        self._ordered = ordered if ordered is not sentinel else None
+        self._ordered_from_sentinel = ordered is sentinel
```

Comment:
This is a little bit gross but seemed to be the cleanest way to implement things.

Internal code can't reference `.ordered` because it could trigger the warning due to under the hood operations that the user has no control over (e.g. handling `'category'` as dtype), so internal code needs to reference `._ordered`. Instead of special casing `sentinel` every time `._ordered` is referenced seemed cleaner to add an additional attribute to keep track of things.

I tried to change the internals so that `.ordered` didn't trigger the warning, but I wasn't successfully able to do so, and my failing attempt had already gotten more complex than the `.ordered` --> `._ordered` change, so didn't seem worthwhile to pursue further.

Note that the `.ordered` --> `._ordered` change only applies to `CategoricalDtype`; this change does not appear to be needed to `Categorical` or `CategoricalIndex`, as those will always resolve to a boolean.

I'll volunteer to remove this and revert `_ordered` to `ordered` when we go through with changing the default value in the constructor.

Reply:
> I'll volunteer to remove this and revert _ordered to ordered when we go through with changing the default value in the constructor.

Showtime!